src/cli/deploy.rs

```diff
@@ -115,15 +115,23 @@ pub async fn run(deploy_args: DeployArgs) -> exitcode::ExitCode {
         }
     };
 
-    let local_replicas = validated_config.scaling.desired_replicas;
+    let local_replicas = validated_config
+        .scaling
+        .as_ref()
+        .map(|local_scaling_config| local_scaling_config.desired_replicas);
 
     // Warn if local scaling config differs from remote
-    let has_scaling_config_drift = cage_scaling_config
-        .as_ref()
-        .is_some_and(|config| config.desired_replicas() != local_replicas);
-    if has_scaling_config_drift {
+    let has_scaling_config_drift = cage_scaling_config.as_ref().is_some_and(|config| {
+        local_replicas.is_some_and(|replicas| config.desired_replicas() != replicas)
+    });
+
+    if has_scaling_config_drift && cage_scaling_config.is_some() {
         let remote_replicas = cage_scaling_config.as_ref().unwrap().desired_replicas();
-        log::warn!("Remote scaling config differs from local config. This deployment will apply the local config.\n\nCurrent remote replica count: {remote_replicas}\nLocal replica count: {local_replicas}\n");
+        let local_replicas_count = local_replicas
+            .map(|count| count.to_string())
+            .unwrap_or(String::from("not_set"));
+
+        log::warn!("Remote scaling config differs from local config. This deployment will apply the local config.\n\nCurrent remote replica count: {remote_replicas}\nLocal replica count: {local_replicas_count}\n");
```

Comment:
I assume this can't print when local_replicas is None? The not_set would be confusing here

Reply:
Yeah, added an expect('Infallible') as it's checked to be `Some` in `has_scaling_config_drift`